### PR TITLE
[CON-1172] Address un-addressed review comments

### PR DESCRIPTION
**Pull request by ac101m**
# Changes
- Use state manager class for host enclave interface rather than plain enum.

### CODE REVIEW

**Comment by ac101m** [comment]
Some changes which I failed to commit for the original PR.

File: conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt [hunk 1]
```diff
@@ -88,7 +90,8 @@ class SocketHostEnclaveInterface(private val port: Int) : HostEnclaveInterface()
             /** Start the message receive thread and allow calls to enter. */
             receiveLoopThread.start()
             callGuardSemaphore.release(maxConcurrentCalls)
-            state = State.RUNNING
+
+            stateManager.transitionStateFrom<State.Ready>(to = State.Running)
```
Comment (shamsasari) [inline conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt:94]: You can move this up and replace it with the `checkStateIs`.

File: conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt [hunk 2]
```diff
@@ -109,7 +114,7 @@ class SocketHostEnclaveInterface(private val port: Int) : HostEnclaveInterface()
             sendMessageToEnclave(SocketCallInterfaceMessage.STOP_MESSAGE)
             receiveLoopThread.join()
 
-            state = State.STOPPED
+            stateManager.transitionStateFrom<State.Running>(to = State.Stopped)
```
Comment (shamsasari) [inline conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt:117]: ditto

**Review by shamsasari — COMMENTED** [review]
I guess there's also the "Interface" to "Link" rename?

File: conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt (hunk 1 above)
Comment (ac101m) [reply conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt:94]: Is this wise? The operations in between can fail. Let me know if I'm over-thinking it.

**Comment by ac101m** [comment]
> I guess there's also the "Interface" to "Link" rename?

You want to do that now? It will involve a large number of changes. Comment adjustments, etc.

File: conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt (hunk 1 above)
Comment (shamsasari) [reply conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt:94]: if `start()` fails then you don't want it to be possible to call it again.

**Review by shamsasari — APPROVED** [review]

**Pull request merged**